src/features/blocks/utils/useBlockTable.ts

```diff
@@ -13,53 +13,56 @@ export type TablePagination = {
 export function useBlockTable() {
   const loading = ref(false);
   const selectedTransaction: Ref<TransactionDto[]> = ref([]);
+  const items: Ref<Transaction[]> = ref([]);
   const pagination: Ref<TablePagination> = ref({
     page: 1,
     page_size: 10,
     pages: 1,
     total: 10,
   });
-
-  const items = computed(() => {
+  function fetch() {
     const res = transactionModel.fetchBlocksList(selectedTransaction.value, {
       page: pagination.value.page,
       page_size: pagination.value.page_size,
     });
-
+    items.value = res.data;
     pagination.value = {
       ...res.pagination,
       pages: Math.ceil(res.pagination.total / res.pagination.page_size),
     };
-
-    return res.data;
-  });
+  }
 
   function setTransaction(transaction: TransactionDto[]) {
     selectedTransaction.value = transaction;
+    fetch();
   }
```

Comment:
Instead of triggering re-computation manually (by calling `fetch()`), set up the reactivity in a way so that `fetch` is called automatically whenever a dependency is changed (e.g. using `watch`).

Reply:
Acknowledged and committed 👍🏼